File: jornada-milhas/src/app/core/types/Passagem.ts
import { Companhia } from "./Companhia"
import { Orcamento } from "./Orcamento"
import { UnidadeFederativa } from "./UnidadeFederativa"

export interface Passagem {
  tipo: string,
  precoIda: number,
  precoVolta: number,
  taxaEmbarque: number,
  conexoes: number,
  tempoVoo: number,
  origem: UnidadeFederativa,
  destino: UnidadeFederativa,
  companhia: Companhia,
  dataIda: Date,
  dataVolta: Date | null,
  total: number,
  orcamento: Orcamento[]
}

File: JornadaMilhasApi/JornadaMilhasApi/Models/Passagem.cs
using System.ComponentModel.DataAnnotations;

namespace JornadaMilhasApi.Models;

public record Passagem : IEntity
{
    [Required]
    public required string Tipo { get; init; }
    [Required]
    public int PrecoIda { get; init; }
    [Required]
    public int PrecoVolta { get; init; }
    [Required]
    public int TaxaEmbarque { get; init; }
    [Required]
    public int Conexoes { get; init; }
    [Required]
    public int TempoVoo { get; init; }
    [Required]
    public virtual Guid OrigemId { get; init; }
    [Required]
    public virtual required Estado Origem { get; init; }
    [Required]
    public virtual Guid DestinoId { get; init; }
    [Required]
    public virtual required Estado Destino { get; init; }
    [Required]
    public virtual Guid CompanhiaId { get; init; }
    [Required]
    public virtual required Companhia Companhia { get; init; }
}
